### PR TITLE
run tests on ruby 2.6

### PR DESCRIPTION
**Pull request by TheRoyalTnetennba**
* Set up kokoro's linux tests to include rubies 2.5.5 and 2.6.2
* Set up OSx to test against 2.5.5 for PR's, and 2.3.8, 2.4.5, 2.5.5, and 2.6.2 for merges/nightlies
* Centralize supported ruby versions and use that to generate build scripts and dockerfiles

https://github.com/googleapis/google-cloud-ruby/issues/2825
https://github.com/GoogleCloudPlatform/ruby-docs-samples/issues/390

### CODE REVIEW

File: .kokoro/docker/ruby-multi/Dockerfile [hunk 1]
```diff
@@ -33,5 +33,6 @@ RUN echo 'gem: --no-rdoc --no-ri' >> /.gemrc
 RUN rbenv install 2.3.8 && rbenv global 2.3.8 && gem install bundler --version 1.17.3
 RUN rbenv install 2.4.5 && rbenv global 2.4.5 && gem install bundler --version 1.17.3
 RUN rbenv install 2.5.3 && rbenv global 2.5.3 && gem install bundler --version 1.17.3
+RUN rbenv install 2.6.0 && rbenv global 2.5.3 && gem install bundler --version 1.17.3
```
Comment (blowmage) [inline .kokoro/docker/ruby-multi/Dockerfile]: `rbenv global 2.6.0`

Comment (TheRoyalTnetennba) [reply .kokoro/docker/ruby-multi/Dockerfile]: :man_facepalming:

**Comment by TheRoyalTnetennba** [comment]
Will hold off on merging until the major dependencies are ready for 2.6

**Comment by blowmage** [comment]
We should be able to support Ruby 2.6 now. Both the grpc and google-protobuf gems have releases that support 2.6. I have been able to run the full suite of tests on Ruby 2.6 locally.

File: .kokoro/build.sh [hunk 2]
```diff
@@ -20,7 +20,7 @@ ruby --version
 # https://github.com/bundler/bundler/issues/6154
 export BUNDLE_GEMFILE=
 
-RUBY_VERSIONS=("2.3.8" "2.4.5" "2.5.3")
+RUBY_VERSIONS=("2.3.8" "2.4.5" "2.5.3" "2.6.0")
```
Comment (blowmage) [inline .kokoro/build.sh]: 2.6.1

File: .kokoro/docker/ruby-multi/Dockerfile [hunk 3]
```diff
@@ -33,5 +33,6 @@ RUN echo 'gem: --no-rdoc --no-ri' >> /.gemrc
 RUN rbenv install 2.3.8 && rbenv global 2.3.8 && gem install bundler --version 1.17.3
 RUN rbenv install 2.4.5 && rbenv global 2.4.5 && gem install bundler --version 1.17.3
 RUN rbenv install 2.5.3 && rbenv global 2.5.3 && gem install bundler --version 1.17.3
+RUN rbenv install 2.6.0 && rbenv global 2.6.0 && gem install bundler --version 1.17.3
```
Comment (blowmage) [inline .kokoro/docker/ruby-multi/Dockerfile]: 2.6.1

File: .kokoro/docker/ruby-multi/Dockerfile [hunk 4]
```diff
@@ -33,6 +33,6 @@ RUN echo 'gem: --no-rdoc --no-ri' >> /.gemrc
 RUN rbenv install 2.3.8 && rbenv global 2.3.8 && gem install bundler --version 1.17.3
 RUN rbenv install 2.4.5 && rbenv global 2.4.5 && gem install bundler --version 1.17.3
 RUN rbenv install 2.5.3 && rbenv global 2.5.3 && gem install bundler --version 1.17.3
-RUN rbenv install 2.6.0 && rbenv global 2.6.0 && gem install bundler --version 1.17.3
+RUN rbenv install 2.6.1 && rbenv global 2.6.1 && gem install bundler --version 1.17.3
```
Comment (blowmage) [inline .kokoro/docker/ruby-multi/Dockerfile]: Do we want to upgrade rubygems as well?

File: Rakefile [hunk 5]
```diff
@@ -5,6 +5,8 @@ require "erb"
 require "fileutils"
 require "timeout"
 
+SUPPORTED_RUBY_VERSIONS = ["2.3.8", "2.4.5", "2.5.3", "2.6.1"]
```
Comment (blowmage) [inline Rakefile]: This list is specific to Kokoro, not a statement on what is supported by the gems. Can we rename so the intended purpose is more clear?

```suggestion
KOKORO_RUBY_VERSIONS = ["2.3.8", "2.4.5", "2.5.3", "2.6.1"]
```

File: Rakefile [hunk 6]
```diff
@@ -545,6 +547,7 @@ namespace :kokoro do
   desc "Generate configs for kokoro"
   task :build do
     generate_kokoro_configs
+    update_ruby_versions
```
Comment (blowmage) [inline Rakefile]:
```suggestion
    update_kokoro_ruby_versions
```

File: Rakefile [hunk 7]
```diff
@@ -693,6 +696,20 @@ def generate_kokoro_configs
   end
 end
 
+def update_ruby_versions
```
Comment (blowmage) [inline Rakefile]:
```suggestion
def update_kokoro_ruby_versions
```

File: Rakefile (hunk 5 above)
Comment (TheRoyalTnetennba) [reply Rakefile]: SGTM

File: Rakefile (hunk 6 above)
Comment (TheRoyalTnetennba) [reply Rakefile]: Since I'm now also updating the readme, are you alright with me leaving this as is?

Comment (blowmage) [reply Rakefile]: I think this needs method name is not very descriptive of what it does. I think it needs to be renamed and/or broken up into smaller methods.

I don't know what to think about having a Kokoro configuration affecting the README.

Comment (TheRoyalTnetennba) [reply Rakefile]: I split them out into separate methods.

**Comment by blowmage** [comment]
Any thoughts on the BigQuery failure on Linux?

**Comment by TheRoyalTnetennba** [comment]
> Any thoughts on the BigQuery failure on Linux?

They don't seem super consistent? I'm spreading out the nightly tests out more - linux, osx, and windows are now each separated by an hour. I don't know how much that will help.

**Comment by blowmage** [comment]
i am hesitant to approve this PR until the Linux build is red.

**Comment by blowmage** [comment]
Would it be possible to upgrade this branch to use Ruby 2.6.2 instead of 2.6.1?

File: .kokoro/osx.sh [hunk 8]
```diff
@@ -28,6 +28,7 @@ function set_failed_status {
 }
 
 # Install/Use 2.5.5
+source /Users/kbuilder/.rvm/scripts/rvm
```
Comment (blowmage) [inline .kokoro/osx.sh]: Does this mean we can also install and build against 2.3, 2.4, and 2.6?

Comment (TheRoyalTnetennba) [reply .kokoro/osx.sh]: It'll take forever, and my understanding is that the mac resources are much more limited than the GCP images. What if we tested all versions for nightly?

**Comment by blowmage** [comment]
🎉  for passing linux build!

**Comment by blowmage** [comment]
So close! 

> Required ruby-2.5.5 is not installed.
> To install do: 'rvm install "ruby-2.5.5"'

**Comment by TheRoyalTnetennba** [comment]
:tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada: :tada:

**Comment by quartzmo** [comment]
From the passing Kokoro CI log:

```sh
ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-linux]
Fetching gem metadata from https://rubygems.org/........
Resolving dependencies....
```

Congratulations!!

**Pull request merged**